io/epoll.cpp

```diff
@@ -301,7 +301,9 @@ ok:     entry.interests |= eint;
             return rm_interest({fd, EVENT_RWE| ONE_SHOT, 0}); // remove fd from epoll
         int ret = add_interest({fd, interest | ONE_SHOT, CURRENT});
         if (ret < 0) LOG_ERROR_RETURN(0, -1, "failed to add event interest");
-        ret = thread_usleep(timeout);
+        // if timeout is just simple 0, wait for a tiny little moment
+        // so that events can be collect.
+        ret = thread_usleep(timeout.timeout() ? timeout : Timeout(10));
```

Comment:
In case of timeout == 0, we'd better reap the events directly without blocking, so as to conform to the semantics.

Reply:
done